### PR TITLE
Nerf pinpointers

### PR DESCRIPTION
**Pull request by Mnemotechnician**
## About the PR
#description is outdated - see the comments

Adds a maximum range to pinpointers, which for normal pinpointers is 128m. When the target entity is more than 128m away, you can observe the same effect as when the tracked entity disappears - the arrow gets "stuck", and if you toggle the pinpointer off and on, it will show a question mark. It will retain this state until the target is in the range again, or until you reconfigure it.

The maximum distance is configurable, and there's also a new [unlimited] variant meant for admemes.

## Why / Balance
Powergamers such as prosmasher use pinpointers to permamently track people and ships. Security can pinpoint *potential* criminals even before they commit anything bad, and criminals can pinpoint security to know if they're being chased. In both cases there's NO COUNTERPLAY to that: once you're pinpointed, your location is forever known.

## Technical details
Added a new if statement to PinpointerSystem.UpdateDirectionToTarget; added a new field to PinpointerComponent and mentioned it in yml.

## Media
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
None

**Changelog**
:cl:
- tweak: Pinpointers now take time to change their target, throughout which it must stand absolutely still, and can no longer be used to pinpoint people.

### CODE REVIEW

**Review by dvir001 — APPROVED** [review]
Working as intended, made a small YML edit to match with our formatting.
@Cheackraze waiting for your final call to merge this code.

**Comment by Cheackraze** [comment]
this will make pinpointers useless for their intended design. Like stated in discord, it should instead have some other change to make them more complicated to use. Make them not work on mobs, give it a do-after time to register to an item. Limiting it this way just removes its core functionality.

**Comment by Mnemotechnician** [comment]
> this will make pinpointers useless for their intended design. Like stated in discord, it should instead have some other change to make them more complicated to use. Make them not work on mobs, give it a do-after time to register to an item. Limiting it this way just removes its core functionality.

Okay, I did both of those. They can no longer pinpoint entities with the MobStateComponent, and have a 15 second long do-after (which breaks upon move of either of the entities, or any fiddling with the pinpointer). The old limiting code is still here - but all pinpointers have unlimited range by default now. It may be useful later. The debug variant can target mobs and has no do-after (specifically, it's 0 seconds long).

**Review by Cheackraze — APPROVED** [review]

**Pull request merged**
